hydromt_delft3dfm/dflowfm.py

```diff
@@ -1431,45 +1431,16 @@ def setup_1dboundary(
             By default 0.1, a small snapping is applied to avoid precision errors.
         """
         self.logger.info(f"Preparing 1D {boundary_type} boundaries for {branch_type}.")
-        boundaries = workflows.get_boundaries_with_nodeid(
-            self.branches,
-            mesh_utils.network1d_nodes_geodataframe(self.mesh_datasets["network1d"]),
-        )
-        refdate, tstart, tstop = self.get_model_time()  # time slice
 
         # 1. get potential boundary locations based on branch_type and boundary_type
         boundaries_branch_type = workflows.select_boundary_type(
-            boundaries, branch_type, boundary_type, boundary_locs
+            self.boundaries, branch_type, boundary_type, boundary_locs
         )
 
         # 2. read boundary from user data
-        if boundaries_geodataset_fn is not None:
-            da_bnd = self.data_catalog.get_geodataset(
-                boundaries_geodataset_fn,
-                geom=boundaries.buffer(
-                    snap_offset
-                ),  # only select data close to the boundary of the model region
-                variables=[boundary_type],
-                time_tuple=(tstart, tstop),
-                crs=self.crs.to_epsg(),  # assume model crs if none defined
-            ).rename(boundary_type)
-            # error if time mismatch
-            if np.logical_and(
-                pd.to_datetime(da_bnd.time.values[0]) == pd.to_datetime(tstart),
-                pd.to_datetime(da_bnd.time.values[-1]) == pd.to_datetime(tstop),
-            ):
-                pass
-            else:
-                self.logger.error(
-                    "forcing has different start and end time. Please check the forcing file. support yyyy-mm-dd HH:MM:SS. "
-                )
-            # reproject if needed and convert to location
-            if da_bnd.vector.crs != self.crs:
-                da_bnd.vector.to_crs(self.crs)
-        elif boundaries_timeseries_fn is not None:
-            raise NotImplementedError()
-        else:
-            da_bnd = None
+        _, da_bnd = self._read_forcing_geodataset(
```

Comment:
I agree that it would be easier for boundary and laterals to work in a more modular way. then we might need to seperate not only constant, geodataframe and geodataset, but also how to represent it internally so we can write into different format again for different situations. e.g. 1D lateral at points needs branchid and chainage, 1D lateral at polygons requires x and y coordinates; 2D boundary for 1D2D applications requires a line and a single timeseries; 2D boundary for 2D3D applications requires a line and multiple timeseries. Good to discuss that in a seperate issue and get them clearly sorted!